govcd/system.go

```diff
@@ -687,6 +687,20 @@ func getExtension(client *Client) (*types.Extension, error) {
 	return extensions, err
 }
 
+// GetStorageProfileByHref fetches storage profile using provided HREF.
+func GetStorageProfileByHref(vcdClient *VCDClient, url string) (*types.VdcStorageProfile, error) {
+
+	vdcStorageProfile := &types.VdcStorageProfile{}
+
+	_, err := vcdClient.Client.ExecuteRequest(url, http.MethodGet,
+		"", "error retrieving storage profile: %s", nil, vdcStorageProfile)
+	if err != nil {
+		return &types.VdcStorageProfile{}, err
```

Comment:
We should return `nil` when a function returning a pointer fails.

Reply:
fixed